tsl/src/reorder.c

```diff
@@ -148,12 +150,45 @@ tsl_move_chunk(PG_FUNCTION_ARGS)
 				 errmsg("valid chunk, destination_tablespace, and index_destination_tablespaces "
 						"are required")));
 
-	reorder_chunk(chunk_id,
-				  index_id,
-				  verbose,
-				  wait_id,
-				  destination_tablespace,
-				  index_destination_tablespace);
+	chunk = ts_chunk_get_by_relid(chunk_id, false);
+
+	if (NULL == chunk)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("\"%s\" is not a chunk", get_rel_name(chunk_id))));
+
+	if (ts_chunk_contains_compressed_data(chunk, &chunk_parent))
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("cannot directly move internal compression data"),
+				 errdetail("Chunk \"%s\" contains compressed data for chunk \"%s\" and cannot be "
+						   "moved directly.",
+						   get_rel_name(chunk_id),
+						   get_rel_name(chunk_parent->table_id)),
+				 errhint("Moving chunk \"%s\" will also move the compressed data.",
+						 get_rel_name(chunk_parent->table_id))));
+
+	/* If chunk is compressed move it by altering tablespace on both chunks */
+	if (OidIsValid(chunk->fd.compressed_chunk_id))
+	{
+		Chunk *compressed_chunk = ts_chunk_get_by_id(chunk->fd.compressed_chunk_id, true);
```

Comment:
For this case, do we simply ignore the index parameter given by the user. Is that parameter optional or always given?

I am thinking that we might want to raise a NOTICE that the reordering could not happen because the chunk was compressed whenever the index parameter is given.

Reply:
index is required on first call, if we decide to print a message here then it should be DEBUG1 or similar